service/mongodb/controller.go

```diff
@@ -163,19 +164,54 @@ func (c Controller) GetFilePermissions(ctx context.Context,
 // otherwise returns nil and any error if occurred.
 func (c Controller) GetUserPermissions(
 	ctx context.Context,
-	userID string) ([]*pb.GetUserPermissionsResponse_FileRole, error) {
-	filter := bson.D{
+	userID string, pageNum int64, pageSize int64, isShared bool, appID string) (*pb.GetUserPermissionsResponse, error) {
+
+	var filter bson.D
+
+	filter = append(filter, bson.E{
+		Key:   PermissionBSONUserIDField,
+		Value: userID,
+	})
+
+	if isShared {
+		filter = append(filter, bson.E{
+			Key:   PermissionBSONCreatorField,
+			Value: bson.M{"$ne": userID},
+		})
+	}
+
+	if appID != "" {
```

Comment:
Rather than verifying if the appID parameter is an empty string (""), you can change this parameter to an optional parameter and check if it is nil - not required, but I think this way is more precise.

Reply:
since it's a string, you cannot compare it to nil, and the parameter is optional by default so it will be the empty string if nothing is received